### PR TITLE
Readability & simple bugs fixes

### PR DESCRIPTION
**Pull request by jkramarz**
should not change actual behaviour (but I don't have any mcHF to test it yet); see commit messages

EDIT: I've ran cppcheck over code base, so no longer only indendation reworks & code deduplication is in this PR, but also few buffer overflows and missing statements fixes.

### CODE REVIEW

File: mchf-eclipse/drivers/ui/lcd/ui_lcd_hy28.c [hunk 1]
```diff
@@ -341,11 +341,9 @@ void UiLcdHy28_SendByteSpi(uint8_t byte)
 }
 void UiLcdHy28_SendByteSpiFast(uint8_t byte)
 {
-
    while ((SPI2->SR & (SPI_I2S_FLAG_TXE)) == (uint16_t)RESET);
    SPI2->DR = byte;
    while ((SPI2->SR & (SPI_I2S_FLAG_RXNE)) == (uint16_t)RESET);
-   byte = SPI2->DR;
```
Comment (jkramarz) [inline mchf-eclipse/drivers/ui/lcd/ui_lcd_hy28.c]: Maybe something was missing here, but this statement has no effects.

File: mchf-eclipse/drivers/ui/lcd/ui_lcd_hy28.c [hunk 2]
```diff
@@ -1692,23 +1690,25 @@ void UiLcdHy28_ShowStartUpScreen(ulong hold_time)
...
-   else
-	   sprintf(tx,"LCD: Parallel Mode");
+
    //
    UiLcdHy28_PrintText(88,150,tx,Grey1,Black,0);
 
    //
    // Display startup frequency of Si570, By DF8OE, 201506
    //
-   int vorkomma = (int)(os.fout);
```
Comment (jkramarz) [inline mchf-eclipse/drivers/ui/lcd/ui_lcd_hy28.c]: There's %u in line 1713

File: mchf-eclipse/drivers/ui/lcd/waterfall_colours.h [hunk 3]
```diff
@@ -499,76 +364,4 @@ const ushort waterfall_rainbow[]	=	{
 	RGB(0xff,0x0e,0x09),
 	RGB(0xff,0xff,0xff),
 };
-
-//
-// DEPRICATED - the "dark" colors are too bright!
```
Comment (jkramarz) [inline mchf-eclipse/drivers/ui/lcd/waterfall_colours.h]: We've Git repo - we can always go back in time, so there's no use of dead code.

File: mchf-eclipse/drivers/ui/misc/hamm_wnd.h [hunk 4]
```diff
@@ -1,19 +0,0 @@
...
-**                                                                                 **
-**  File name:                                                                     **
-**  Description:                                                                   **
-**  Last Modified:                                                                 **
-**  Licence:		For radio amateurs experimentation, non-commercial use only!   **
-************************************************************************************/
-
-#ifndef __HAMM_WND_H
-#define __HAMM_WND_H
-
```
Comment (jkramarz) [inline mchf-eclipse/drivers/ui/misc/hamm_wnd.h]: Just an empty Hamming window? :-)

File: mchf-eclipse/drivers/ui/oscillator/ui_si570.c [hunk 5]
```diff
@@ -524,24 +525,30 @@ uchar ui_si570_set_frequency(ulong freq,int calib,int temp_factor, uchar test)
...
+		return(2);
+	}
+	else
+	{
+		// tuning outside hard limits, tuning bad, color red
+		if (si_freq > SI570_MAX_FREQ / 1000000)
+			si_freq = SI570_MAX_FREQ / 1000000;
+		if (si_freq < SI570_MIN_FREQ / 1000000)
+			si_freq = SI570_MIN_FREQ / 1000000;
+		ui_si570_change_frequency(si_freq, test);
```
Comment (jkramarz) [inline mchf-eclipse/drivers/ui/oscillator/ui_si570.c]: The only real change in this file, AFAIR.

File: mchf-eclipse/drivers/ui/oscillator/ui_si570.c [hunk 6]
```diff
@@ -659,43 +666,60 @@ void ui_si570_conv_temp(uchar *temp,int *dtemp)
 //* Functions called    :
 //*----------------------------------------------------------------------------
 //
-void ui_si570_calc_startupfrequency(void)
+void ui_si570_calculate_startup_frequency(void)
 {
-if (os.fout < 5)
-	{
+	if(os.fout >= 5)
```
Comment (jkramarz) [inline mchf-eclipse/drivers/ui/oscillator/ui_si570.c]: Oh, OK. Here's the second one.

File: mchf-eclipse/drivers/ui/ui_driver.c [hunk 7]
```diff
@@ -9950,7 +9834,7 @@ void UiCheckForPressedKey(void)
 	bool stat = 1;
 	poweroffcount = rbcount = 0;
 	p_o_state = rb_state = new_state = 0;
-	char txt[32];
+	char txt[40];
```
Comment (jkramarz) [inline mchf-eclipse/drivers/ui/ui_driver.c]: "press & hold POWER-button to poweroff" - it's 38 chars

File: mchf-eclipse/hardware/mchf_board.c [hunk 8]
```diff
@@ -733,68 +733,65 @@ void mchf_board_switch_tx(char mode)
 void mchf_board_power_off(void)
 {
 	ulong i;
-	char	tx[32];
+	char	tx[40];
```
Comment (jkramarz) [inline mchf-eclipse/hardware/mchf_board.c]: Again, one of the strings is longer. I wonder what got corrupted by this call :-)

File: mchf-eclipse/hardware/mchf_board.h [hunk 9]
```diff
@@ -1335,7 +1335,7 @@ typedef struct TransceiverState
 
 	uchar	scope_trace_colour;	// color of spectrum scope trace;
 	uchar	scope_grid_colour;	// saved color of spectrum scope grid;
-	ulong	scope_grid_colour_active;	// active color of spectrum scope grid;
+	ushort	scope_grid_colour_active;	// active color of spectrum scope grid;
```
Comment (jkramarz) [inline mchf-eclipse/hardware/mchf_board.h]: It seemed to be inconsitent with `#define RGB(red,green,blue)(uint16_t)(((red>>3)<<11)|((green>>2)<<5)|(blue>> 3))
`

File: mchf-eclipse/hardware/mchf_hw_i2c2.c [hunk 10]
```diff
@@ -265,24 +268,29 @@ uint16_t Write_24Cxxseq(uint32_t Addr, uint8_t *buffer, uint16_t length, uint8_t
...
-		page = 64;
-	if(Mem_Type == 13)
-		page = 32;
-	if(Mem_Type == 14 || Mem_Type == 15)
-		page = 64;
-	if(Mem_Type > 15 && Mem_Type < 19)
-		page = 128;
-	if(Mem_Type == 19)
-		page = 256;
-
```
Comment (jkramarz) [inline mchf-eclipse/hardware/mchf_hw_i2c2.c]: Otherwise, let's use page=random :-)

Comment (df8oe) [inline mchf-eclipse/hardware/mchf_hw_i2c2.c]: No other Mem_Types like the above are possible so "otherwise" is without any meaning.

**Comment by df8oe** [comment]
There are changes I see 100% they fix bugs or improve code speed/readability etc. Others are a "can do it" - some people say its better like the old way and others its better the new way. But there are others I have doubt if they could break functionality. So I want to review the changes manually and push them to devel-DF8OE. That will take some time because I do not want to hurry this work.

It will be great if you can finish the building of your mcHF so that you can test code functionality which is not 100% sure before you commit them.... Of course I apreciate your work and will implement ASAP.

Many thanks again.

At this point I hope I can implement the changes at the same speed you commit new ones otherwise we will get a delay ;)

vy 73
Andreas

**Comment by jkramarz** [comment]
Ok. Please use git cherry-pick when introducing these changes to
repository, I'd prefer to keep git author this time ;-)

**Comment by df8oe** [comment]
I will do so.

vy 73
Andreas

**Comment by jkramarz** [comment]
If you prefer to, I can split this PR into two, taking only the later changes in the new one

**Comment by df8oe** [comment]
I have cherry-picked all that seems ok

**Comment by df8oe** [comment]
Now I need a *BIG* cup of coffee :)

vy 73
Andreas

**Comment by jkramarz** [comment]
Thank you for testing these changes for me, Andreas!

**Comment by df8oe** [comment]
Hello Jakub,

welcome in our development community! Great work without having a mcHF for testing.... Your efforts are appreciated and I hope we can/will mprove firmware in future much more ;)

vy 73
Andreas

**Pull request closed without merging**